templates/CRM/Mosaico/Page/EditorIframe.tpl

```diff
@@ -24,6 +24,12 @@
 
       var plugins = [];
       var config = {/literal}{$mosaicoConfig}{literal};
+
+      window.onbeforeunload = function(e) {
+        e.preventDefault();
+        e.returnValue = "Do you really want to leave our brilliant application?";
```

Comment:
This message is funny but could be a little more informative. And translated. How about this:

```suggestion
        e.returnValue = "{/literal}{ts}Exit email composer without saving?{/ts}{literal}";
```

Reply:
Actually, I'm not sure because I haven't tried it... does this message pop up regardless of unsaved changes? Does it even pop up when exiting the application normally? That might be a problem.

Reply:
Haha! The message doesn't actually appear at all in most modern browsers which is why I didn't notice I'd left it per an example I found... Totally agree on updating it per your suggestion. It won't popup when exiting normally because it's triggered by the browser when clicking something outside the context of the form/window that will trigger it to disappear.